src/coreclr/jit/codegencommon.cpp

```diff
@@ -4118,7 +4118,37 @@ void CodeGen::genEnregisterOSRArgsAndLocals()
 
         genInstrWithConstant(ins_Load(lclTyp), size, varDsc->GetRegNum(), genFramePointerReg(), offset, initReg);
         *pInitRegZeroed = false;
-#endif // TARGET_ARM64 || TARGET_LOONGARCH64 || TARGET_RISCV64
+#elif defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
+
+        // Patchpoint offset is from top of Tier0 frame
+        //
+        // We need to determine the frame-pointer relative
+        // offset for this variable in the osr frame.
+        //
+        // First there is no need to ajust stkOffs
+        // as it relative to sp within Tier0 frame
+
+        // then add the OSR frame size
+        //
+        const int osrFrameSize = genTotalFrameSize();
+
+        // then subtract OSR SP-FP delta
+        //
+        const int osrSpToFpDelta = genSPtoFPdelta();
+
+        //                 | => tier0 top of frame relative
+        //                 |         + => osr bottom of frame (sp) relative
+        //                 |         |              - => osr fp relative
+        //                 |         |              |
+        const int offset = stkOffs + osrFrameSize - osrSpToFpDelta;
+
+        JITDUMP("---OSR--- V%02u (reg) Tier0 virtual offset %d OSR frame size %d OSR sp-fp "
+                "delta %d total offset %d (0x%x)\n",
+                varNum, stkOffs, osrFrameSize, osrSpToFpDelta, offset, offset);
+
+        genInstrWithConstant(ins_Load(lclTyp), size, varDsc->GetRegNum(), genFramePointerReg(), offset, initReg);
+        *pInitRegZeroed = false;
+#endif // TARGET_LOONGARCH64 || TARGET_RISCV64
```

Comment:
Is any of this code necessary? Can we share LA64/RISCV64 OSR logic remain shared with ARM64 by changing
```
const int offsetAdjust = codeGen->genSPtoFPdelta();
```
to
```
const int offsetAdjust = codeGen->genSPtoFPdelta() - totalFrameSize;
```
in tier0?

Reply:
IMO, I think the ARM64's OSR is not so good here.
Maybe we should amend these code by a new PR ?

Reply:
> IMO, I think the ARM64's OSR is not so good here. Maybe we should amend all related OSR code by a new PR ?

For ARM64's OSR, first sub `info.compPatchpointInfo->TotalFrameSize()` on the line-6578 of `Compiler::lvaAssignVirtualFrameOffsetsToLocals`, then later add `info.compPatchpointInfo->TotalFrameSize()` on line-5719 of `Compiler::lvaFixVirtualFrameOffsets`.

Reply:
> I would prefer we keep ARM64/LA64/RISCV64 shared/uniform in this PR if we can.
> 
> I'd be fine with changing the offsets to be relative to something else in a new PR if that simplifies things. It would make sense for me if we made them relative to the bottom of the tier0 stack frame on all platforms, but it would be @AndyAyersMS's decision about the change.
> 
> Currently I'm not sure I understand what exactly the encoded offsets are relative to on the different platforms. Perhaps @AndyAyersMS can shed some light on this?

I agree with you.
It's best to discuss with @AndyAyersMS to get better design.

Reply:
@jakobbotsch 
I had push a new patch to uniform the ARM64 to LA64/RV64.

Reply:
The change to arm64 will need JIT-EE GUID update. I think x64 can also be changed such that offsets are always relative to the bottom of the frame for all targets. But I think it would be better to do this as a separate PR instead of doing it in this PR. For this PR it's best if you leave arm64/x64 unchanged and keep LA64/RISCV64 similar to arm64.

Reply:
Ok, thanks

Reply:
> Currently I'm not sure I understand what exactly the encoded offsets are relative to on the different platforms. Perhaps @AndyAyersMS can shed some light on this?

At least for x64 and arm64, they are offsets are relative to the top of the Tier0 frame. 

How does (the idea of) changing this make things simpler?

Reply:
OK, thanks.
If needed, maybe it's better to discuss this in details by a new PR later.